concepts/error-handling/about.md

```diff
@@ -12,69 +11,81 @@ There are multiple ways to handle errors in `Cairo`.
 
 ## The Result Enum
-This approach handles errors gracefully, providing feedback insted of crashing or producing undefined behaviiour.
+This approach handles errors gracefully, providing feedback insted of crashing or producing undefined behaviour.
 
-+ Let's look at an example of `Divide By Zero` :
+```
+enum Result<T, E> {
+    Ok: T,
+    Err: E,
+}
+```
+
+Let's look at an example of `Divide By Zero`:
 ```rust
-fn divide(a: felt252, b: felt252) -> Result<felt252, ByteArray> {
+fn divide(a: u32, b: u32) -> Result<u32, ByteArray> {
     if b == 0 {
-        return Err("Error: Division by zero"); // Return an error if division by zero is attempted
+        return Result::Err("Error: Division by zero"); // Return an error if division by zero is attempted
     }
-    Ok(a / b)
+    Result::Ok(a / b)
 }
 
 fn main() {
     // Test with division by zero
     match divide(10, 0) {
-        Ok(result) => println!("Division successful: {}", result),
-        Err(e) => println!("{}", e),
+        Result::Ok(result) => println!("Division successful: {}", result),
+        Result::Err(e) => println!("{}", e),
     }
 }
 ```
 ## The `?` Operator
 `?` operator is used for concise and implicit error handling, letting the calling function deal with any errors that might occur.
 
-+ Example
+Example
 ```rust
-fn safe_divide(a: felt252, b: felt252) -> Result<felt252, ByteArray> {
+fn safe_divide(a: u32, b: u32) -> Result<felt252, ByteArray> {
     // The ? operator automatically generate the error if it occurs
     let result = divide(a, b)?;
-    Ok(result)
+    Result::Ok(result)
 }
 
 fn main() {
     match safe_divide(10, 0) {
-            Ok(value) => println!("Division successful: {}", value),
-            Err(e) => println!("{}", e),
+        Ok(value) => println!("Division successful: {}", value),
+        Err(e) => println!("{}", e),
     }
 }
 ```
 
 ## Error with `panic` 
 When program encounters an error which is unrecoverable then a `panic` occurs. This will stop the execution, which is helpful in the case where continuing the execution wouldn't make any sense.
 
-+ Example
+Example
 ```rust
 fn main() {
-    let mut data = array![1, 2];
+    let data = array![1, 2];
 
     if true {
         panic(data);
     }
     println!("This line will not be reached.");
 }
 ```
+Running this example would print this in the console:
+
+```console
+Run panicked with [1, 2, ].
+```
 
-> `panic!` macro is panic error that allows you to use a string as error message (can include longer than 31 bytes) accepts ByteArrays as argument. 
+There is also a special macro called `panic!`, that allows you to use a `ByteArray` as the error message.
 ```rust
-panic!("the error for panic! macro is not limited to 31 characters anymore");
+panic!("The error for the panic! Error message is not limited to 31 characters anymore");
 ```
 
 ## `nopanic` Notation
 Cairo `nopanic` notation indicates that a function will never panic. It makes code more user-friendly. 
```

Comment:
Don't forget to [run the `./bin/lint_markdown.sh` command in you terminal](https://github.com/exercism/cairo/pull/131#discussion_r1730300530)